external/storm-kafka/src/jvm/storm/kafka/KafkaDataStore.java

```diff
@@ -0,0 +1,219 @@
+package storm.kafka;
+
+import com.google.common.collect.Maps;
+import kafka.api.ConsumerMetadataRequest;
+import kafka.common.ErrorMapping;
+import kafka.common.OffsetAndMetadata;
+import kafka.common.OffsetMetadataAndError;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.ConsumerMetadataResponse;
+import kafka.javaapi.OffsetCommitRequest;
+import kafka.javaapi.OffsetCommitResponse;
+import kafka.javaapi.OffsetFetchRequest;
+import kafka.javaapi.OffsetFetchResponse;
+import kafka.network.BlockingChannel;
+import org.json.simple.JSONValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class KafkaDataStore implements StateStore {
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaDataStore.class);
+
+    private SpoutConfig _spoutConfig;
+    private Partition _partition;
+
+    private String _consumerGroupId;
+    private String _consumerClientId;
+    private int _stateOpTimeout;
+    private int _stateOpMaxRetry;
+
+    private int _correlationId = 0;
+    private BlockingChannel _offsetManager;
+
+    public KafkaDataStore(Map stormConf, SpoutConfig spoutConfig, Partition partition) {
```

Comment:
you shouldn't need <tt>stormConf</tt> or <tt>spoutConfig</tt> to instantiate.  I believe you would just need:
- broker connect (host:port)
- consumerId
- optional topic? not sure

Your store should support all partitions, so leave out <tt>partition</tt>.